config.hh

```diff
@@ -234,6 +234,9 @@ private:
   // in XFCE with Qt4. Let us enable customization code for Qt5 exclusively to
   // avoid regressions with Qt4.
   #define ENABLE_SPWF_CUSTOMIZATION
+
+  // Whether to XDG Base Directory specification might be followed
+  #define XDG_BASE_DIRECTORY_COMPLIANCE
```

Comment:
`ENABLE_SPWF_CUSTOMIZATION` is defined in this header because it is used in multiple source files. `XDG_BASE_DIRECTORY_COMPLIANCE` is a completely independent variable. It is limited to Linux/BSD and Qt5 for different reasons, so there is no semantic benefit to lumping the two variables together. Since the variable is currently used in a single source file, and I don't see why it should be needed elsewhere, I believe it should be defined in the source file. This will simplify reasoning about the code, similarly to how unnamed namespaces are used.

The comment above the variable should explain why it is not defined in Qt4: requires a separate implementation and there is little motivation to write and test code for obsolete framework versions.

Reply:
`XDG_BASE_DIRECTORY_COMPLIANCE` is indeed not used in other files, but I think it makes sense to define it here, to avoid another #if in a `config.cc` that is already > 2k long. I would say that disadvantage of it having defined in other files is amortized by that. Following your argumentation, I could argue that `ENABLE_SPWF_CUSTOMIZATION` perhaps shouldn't be defined in `config.hh` as well. It's not used in `config.cc`, it's only used in two source files that don't include `config.hh` directly, and `config.hh` is included directly in 43 files. Probably it would be semantically best if that define was in separate header file that is included only in these two source files, but it's fine as it is.

If you insist, I will move define to source file, but I think that current define is a reasonable trade-off.

Reply:
> Probably it would be semantically best if that define was in separate header file that is included only in these two source files, but it's fine as it is.

Creating a separate header for one preprocessor constant would be an overkill. Besides, `ENABLE_SPWF_CUSTOMIZATION` is defined just above related declarations, which are only useful in builds where `ENABLE_SPWF_CUSTOMIZATION` is defined. So in this case the definition also serves documentation purpose. In principle, the code below that definition could be `#ifdef`-ed as well, but that would require more `#ifdef`s in other files.

> to avoid another #if in a config.cc that is already > 2k long.

Well, config.cc (along with many other files) includes config.hh. Saving a couple of lines in the header means less work for the compiler and for code readers (who probably read the header code more often than the source file code). And again, it saves the readers from wondering where else is this preprocessor definition used. So yes, if the length of *config.cc* is your main argument, then I insist on moving it.